dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/DatadogCore.kt

```diff
@@ -197,6 +198,11 @@ internal class DatadogCore(
                 val mutableContext = featureContext.toMutableMap()
                 updateCallback(mutableContext)
                 it.setFeatureContext(featureName, mutableContext)
+                // notify all the other features
+                features.filter { it.key != featureName }
+                    .forEach { (_, feature) ->
+                        feature.notifyContextUpdated(featureName, mutableContext)
```

Comment:
even though the type on the receiver side is `Map` here, maybe it is worth to make a copy as a safe-guard?

```suggestion
                        feature.notifyContextUpdated(featureName, mutableContext.toMap())
```